.github/workflows/maven-central-publish.yml

```diff
@@ -0,0 +1,145 @@
+            target: aarch64-apple-darwin
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+          - os: windows-latest
+            target: aarch64-pc-windows-msvc
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: "21"
+          distribution: "temurin"
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          targets: ${{ matrix.target }}
+
+      - name: Install cross-compilation tools
+        if: matrix.target == 'aarch64-unknown-linux-gnu'
+        run: sudo apt-get update && sudo apt-get install -y gcc-aarch64-linux-gnu
+
+      - name: Build the bindings
+        run: make kotlin TARGET=${{ matrix.target }}
+        env:
+          CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER: aarch64-linux-gnu-gcc
+
+      - name: Upload native library
+        uses: actions/upload-artifact@v4
+        with:
+          name: native-lib-${{ matrix.os }}-${{ matrix.target }}
+          path: bindings/kotlin/lib/**/*iota_sdk_ffi*
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: "21"
+          distribution: "temurin"
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Download native libraries
+        uses: actions/download-artifact@v4
+        with:
+          path: libs
+
+      - name: Prepare libraries
+        run: |
+          echo "Listing contents of libs directory:"
+          ls -la libs/
+          echo "Listing contents of native-lib-ubuntu-latest-x86_64-unknown-linux-gnu:"
+          ls -la libs/native-lib-ubuntu-latest-x86_64-unknown-linux-gnu/
+          echo "Listing contents of bindings/kotlin/lib before copying:"
+          ls -la bindings/kotlin/lib/
+          cd bindings/kotlin/lib
+          echo "Creating platform directories and copying libraries..."
+          mkdir -p linux-x86-64 linux-aarch64 darwin-x86-64 darwin-aarch64 win32-x86-64 win32-aarch64
+          cp ../../../libs/native-lib-ubuntu-latest-x86_64-unknown-linux-gnu/linux-x86-64/libiota_sdk_ffi.so linux-x86-64/
+          cp ../../../libs/native-lib-ubuntu-latest-aarch64-unknown-linux-gnu/linux-aarch64/libiota_sdk_ffi.so linux-aarch64/
+          cp ../../../libs/native-lib-macos-latest-x86_64-apple-darwin/darwin-x86-64/libiota_sdk_ffi.dylib darwin-x86-64/
+          cp ../../../libs/native-lib-macos-latest-aarch64-apple-darwin/darwin-aarch64/libiota_sdk_ffi.dylib darwin-aarch64/
+          cp ../../../libs/native-lib-windows-latest-x86_64-pc-windows-msvc/win32-x86-64/iota_sdk_ffi.dll win32-x86-64/
+          cp ../../../libs/native-lib-windows-latest-aarch64-pc-windows-msvc/win32-aarch64/iota_sdk_ffi.dll win32-aarch64/
```

Comment:
I think you can completely get rid of these cps, have a look at `actions/download-artifact`'s `merge-multiple` and maybe my workflow, I had the same situation at some point